pilot/pkg/credentials/kube/multicluster.go

```diff
@@ -16,92 +16,54 @@ package kube
 
 import (
 	"fmt"
-	"sync"
 
 	"istio.io/istio/pilot/pkg/credentials"
 	"istio.io/istio/pkg/cluster"
 	"istio.io/istio/pkg/kube/multicluster"
-	"istio.io/istio/pkg/log"
 )
 
 // Multicluster structure holds the remote kube Controllers and multicluster specific attributes.
 type Multicluster struct {
-	remoteKubeControllers map[cluster.ID]*CredentialsController
-	m                     sync.Mutex // protects remoteKubeControllers
-	configCluster         cluster.ID
-	secretHandlers        []func(name string, namespace string)
+	configCluster  cluster.ID
+	secretHandlers []func(name string, namespace string)
+	component      *multicluster.Component[*CredentialsController]
 }
 
 var _ credentials.MulticlusterController = &Multicluster{}
 
-func NewMulticluster(configCluster cluster.ID) *Multicluster {
+func NewMulticluster(configCluster cluster.ID, controller multicluster.ComponentBuilder) *Multicluster {
 	m := &Multicluster{
-		remoteKubeControllers: map[cluster.ID]*CredentialsController{},
-		configCluster:         configCluster,
+		configCluster: configCluster,
 	}
 
+	m.component = multicluster.BuildMultiClusterComponent(controller, func(cluster *multicluster.Cluster) *CredentialsController {
+		return NewCredentialsController(cluster.Client, m.secretHandlers)
```

Comment:
secretHandlers are registered from `(m *Multicluster) AddSecretHandler` later, how can that be passed during the initiation

Reply:
Pre-existing ones are added here, then new ones get added later during `AddSecretHandler`.

Reply:
See https://github.com/istio/istio/pull/49260#discussion_r1500073199

Reply:
Resolving to consolidate conversation on https://github.com/istio/istio/pull/49260#discussion_r1500073199